system/ee/legacy/core/Config.php

```diff
@@ -1805,8 +1805,7 @@ function site_url($uri = '')
 				$uri = implode('/', $uri);
 			}
 
-			$suffix = ($this->item('url_suffix') == FALSE) ? '' : $this->item('url_suffix');
-			return $this->slash_item('base_url').$this->item('index_page').'?'.trim($uri, '/').$suffix;
+			return reduce_double_slashes($this->slash_item('base_url') . $this->item('index_page') . trim($uri, '/'));
```

Comment:
This fix looks like it removes the ability to use the URL Suffix?

Reply:
It does, but I'm pretty sure the url_suffix went away.  See https://expressionengine.com/forums/archive/topic/205246/my-url_suffix-is-not-working#958040 though I know that's not where I initially saw it.

Reply:
OK- here's DJ replying to a bug report about url_suffix not working: https://expressionengine.com/have-the-courage-to-pursue.php?/cp/publish/comments/entry/10989

Reply:
It's listed in the docs still - so we'll need to remove it from there if it's not a part of EE currently: https://docs.expressionengine.com/latest/development/legacy/helpers/url-helper.html

Reply:
Cool- I removed from docs- see [pull request 88](https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/88).  Searched docs, that's the only reference that came up.

I also tested.  If I set the config, it had zero effect on the site_url, but interestingly it did allow urls with the suffix to be displayed while without the config, those same urls didn't resolve.

So then I tested with the above change and it didn't affect that behavior, which I take as a good thing.